integration_test/gce/gce_testing.go

```diff
@@ -345,7 +345,9 @@ func isRetriableLookupMetricError(err error) bool {
 	return strings.Contains(err.Error(), "Internal error") ||
 		strings.Contains(err.Error(), "The metric referenced by the provided filter is unknown") ||
 		// workload.googleapis.com/* domain metrics are created on first write, and may not be immediately queryable.
-		strings.Contains(err.Error(), "failed to look up metric workload.googleapis.com")
+		// The error doesn't always look the same, hopefully looking for "code = NotFound" will catch all variations.
+		strings.Contains(err.Error(), "failed to look up metric workload.googleapis.com") ||
+		strings.Contains(err.Error(), "code = NotFound")
```

Comment:
Is it possible to check the code directly instead of checking the string version?

Reply:
(with grouchiness) this api doesn't seem to document what kind of error object is being returned, and looking at the code didn't help either: https://github.com/googleapis/google-cloud-go/blob/monitoring/v1.2.0/monitoring/apiv3/metric_client.go#L362

it could be some kind of gax-ian error object, or it could be decided by c.metricClient.ListTimeSeries, which just links back to itself, very unhelpful.

I agree that matching on the error string is gross but don't have a better solution at the moment. suggestions welcome.

Reply:
I would expect it returns an error compatible with [`status.FromError`](https://pkg.go.dev/google.golang.org/grpc/status#FromError), which returns a `Status` object that has a `Code()` method that you can compare against `codes.NotFound`.

I would also expect "failed to look up metric" to be paired with a code (perhaps even `NotFound`), though maybe you want to explicitly match on `workload.googleapis.com` for some reason?

Reply:
OK I changed it to just look for NotFound (and Internal). Much cleaner and hopefully more robust to tiny changes in the error message. It seems to work.